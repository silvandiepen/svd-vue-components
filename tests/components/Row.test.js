import Vue from "vue/dist/vue.js";
import simulant from "simulant";
import TableComponent from "../../src/";

describe("TableComponent", () => {
  beforeEach(() => {
    localStorage.clear();
  });

  afterEach(() => {
    TableComponent.settings({
      filterNoResults: "There are no matching rows"
    });
  });

  it("can mount", async () => {
    document.body.innerHTML = `
            <div id="app">
                <row center>
                    <column></column>
                </row>
            </div>
        `;

    await createVm();

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it("can display nested properties", async () => {
    document.body.innerHTML = `
            <div id="app">
                <row center>
                    <column>
                        <row center>
                            <column></column>
                        </row>
                    </column>
                </row>
            </div>
        `;

    await createVm();

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});

async function createVm(options = {}) {
  const vm = new Vue({
    el: "#app",
    ...options
  });

  await Vue.nextTick(() => {});

  const rows = vm.$children[0];

  return rows;
}
