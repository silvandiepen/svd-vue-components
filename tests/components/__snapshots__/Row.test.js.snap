// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent can display nested properties 1`] = `

<div id="app">
  <div class="row center">
    <div class="column">
      <div class="row center">
        <div class="column">
        </div>
      </div>
    </div>
  </div>
</div>

`;

exports[`TableComponent can mount 1`] = `

<div id="app">
  <div class="row center">
    <div class="column">
    </div>
  </div>
</div>

`;
